### DISCUSSION TITLE
En qué carpeta se corren los inputs?

### DISCUSSION THREAD

**Discussion by cron53** [IIC2133-PUC/2024-2#42] [Tarea 0] [answered]
Hola quería preguntar exactamente qué path tengo que poner para leer input.txt y output.txt. En mi programa puse los tests en una carpeta, que tiene 9 carpetas de cada test, cada una con input.txt y output.txt. Mi programa concatena el path /executables/part-x-yy-z/. Digo eso en el readme? ¿O los inputs van a estar en el directorio de más afuera? Espero que se entienda, gracias!

**Comment by tomastrivino** [comment, marked as answer]
No entiendo lo de "concatenar". El input se tiene que poner en un path relativo 

Supongamos que tienes la siguiente estructura:

Carpeta tarea
- puffle <binario, archivo que ejecutas como ./puffle>
- src (carpeta donde hiciste la tarea y está el main.c etc)
- tests
   -   easy-1.txt
   -   carpeta 
      -  easy-2.txt

./puffle lo ejecutas desde la carpeta "tarea". Desde esa carpeta, el path relativo a easy-1.txt es tests/easy-1.txt y el path relativo a easy-2.txt es tests/carpeta/easy-2.txt.

Así, para ejecutar el binario con el input de easy-2 sería (asumiendo que quieres el output a out.txt y que estás en la carpeta "tarea" en la consola)
`./puffle tests/carpeta/easy-2.txt out.txt`

**Reply by cron53** [reply]
A lo que me refiero es cómo tengo que dejar entregado mi script. Para que ustedes pongan:
Opción 1: ./puffle executables/part-2-medium-1/input.txt executables/part-2-medium-1/output.txt
Opción 2: ./puffle input.txt output.txt

En esta última opción, van a poner los scripts en Carpeta tarea, o en la carpeta executables. En mi script tengo: 
`char input_path[256];
 char output_path[256];

snprintf(input_path, sizeof(input_path), "excecutables/part-1-easy/%s", argv[1]);
snprintf(output_path, sizeof(output_path), "excecutables/part-1-easy/%s", argv[2]);

FILE *input_file = fopen(input_path, "r");
FILE *output_file = fopen(output_path, "w");
`
Porque tengo esa estructura. Mi pregunta va sólo a dónde van a poner ustedes los tests, y si ponen el path relativo completo, o solo input.txt output.txt. 

Sé que es un detalle muy chico, pero es para dejarla entregada totalmente.

**Reply by sebacarrascop** [reply]
hola, el código base ya tenía funcional la forma en que se leen los archivos ./puffle path_al_input path_al_output
Cualquier cambio que hagas debe ser consistente ya que los test se ejecutan de forma automatizada entregando los paths relativos a los tests, no es relevante donde tu guardes tus tests ya que correremos tests diferentes para la evaluación
